lib/ipaddr.mli

```diff
@@ -54,20 +54,22 @@ module V4 : sig
   (** Converts the low bytes of four int values into an abstract {! V4.t }. *)
   val make : int -> int -> int -> int -> t
 
-  (** {3 Text string conversion} *)
+  (** {3 Text string conversion}
+      These manipulate human-readable IPv4 addresses (for example [192.168.1.2]). *)
 
-  (** [of_string s] is the address {!t} represented by the IPv4 address [s].
-      Returns a human-readable error string if parsing failed. *)
+  (** [of_string s] is the address {!t} represented by the human-readable IPv4
+      address [s]. Returns a human-readable error string if parsing failed. *)
   val of_string : string -> (t, [> `Msg of string ]) result
 
-  (** [of_string_exn s] is the address {!t} represented
-      by [s]. Raises {!Parse_error} if [s] is not a
-      valid representation of an IPv4 address. *)
+  (** [of_string_exn s] is the address {!t} represented as a human-readable IPv4
+      address [s]. Raises {!Parse_error} if [s] is not exactly 4 bytes long. *)
   val of_string_exn : string -> t
 
   (** [of_string_raw s off] acts as {!of_string_exn} but takes as an extra
       argument the offset into the string for reading. [off] will be
-      mutated to an unspecified value during the function call. *)
+      mutated to an unspecified value during the function call. [s] must be
+      of at least length [off+4] or else a {!Parse_error} exception will be
```

Comment:
The `length [off+4]` (and above `not exactly 4 bytes long`) is incorrect, this parses a dotted-quad string representation which is at least 7 characters (up to 15).